xchange-stream-service-netty/src/main/java/info/bitrich/xchangestream/service/netty/NettyStreamingService.java

```diff
@@ -408,12 +408,14 @@ protected String getChannel(T message) {
 
   protected void handleMessage(T message) {
     String channel = getChannel(message);
-    handleChannelMessage(channel, message);
+    if (channel != null)
```

Comment:
Hm, it doesn't look like valid behaviour for `getChannel()` to return `null`; `""` seems to be expected (which is what it does if there's an exception parsing the channel from the message). Perhaps instead change `getChannel()` to catch `null` and replace it with `""`?

Reply:
getChannel is overridden, so there is no knowing what it will return, "" or null or "a channel".  I agree probably dont want to send null or "" to handleChannelMessage.  I will add a check for "";

I added a log for handleError because I dont think we want that to fail silently.